app/services/process_mentions_service.rb

```diff
@@ -10,23 +10,21 @@ class ProcessMentionsService < BaseService
   def call(status)
     return unless status.local?
 
-    status.text.scan(Account::MENTION_RE).each do |match|
-      username, domain  = match.first.split('@')
-      mentioned_account = Account.find_remote(username, domain)
-
-      if mentioned_account.nil? && !domain.nil?
-        begin
-          mentioned_account = resolve_remote_account_service.call(match.first.to_s)
-        rescue Goldfinger::Error, HTTP::Error
-          mentioned_account = nil
-        end
+    status.text = status.text.gsub(Account::MENTION_RE) do |match|
+      begin
+        mentioned_account = resolve_remote_account_service.call($1)
```

Comment:
Codeclimate complains about $1, but I'm not sure there is a better way to do that in Ruby, at least I have not found one reading the documentation for gsub.

Reply:
what's wrong with `resolve_remote_account_service.call(match.first.to_s)`?

Reply:
match is the full matched string, not a list of groups :/